### PR TITLE
Fix race-conditions due to lack of await on element.buildInternal

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

**Comment by samouri** [comment]
Was this race preexisting? How did you find it?

File: test/unit/test-resources.js [hunk 1]
```diff
@@ -45,6 +45,62 @@ describes.realWin('Resources', {amp: true}, (env) => {
...
+      taskId: 'resource#P',
+      ...opts,
+    };
+
+    const element = document.createElement('amp-el');
+    element.V2 = () => false;
+    element.isBuilt = () => isBuilt;
+    element.isBuilding = () => isBuilding;
+
+    const resource = new Resource(id, element, resources);
```
Comment (samouri) [inline test/unit/test-resources.js:83]: nit: (not sure if this is better, but def more concise)
```js
Object.entries(opts, ([prop, val]) => {
  env.sandbox.stub(resources, prop).returns(val);
});
```

**Review by samouri — APPROVED** [review]

**Comment by dvoytenko** [comment]
> Was this race preexisting? How did you find it?

I find it via #32568. It introduced an extra microtask sometimes. However, these tests were not using the API per spec, so I decided to fix them separately.

File: test/unit/test-resources.js (hunk 1 above)
Comment (dvoytenko) [reply test/unit/test-resources.js:83]: Unfortunately, there are some minor naming differences. So I think it's ok for now to stub separately.

**Comment by samouri** [comment]
> I find it via #32568. It introduced an extra microtask sometimes. However, these tests were not using the API per spec, so I decided to fix them separately.

Thank you! I appreciate the hard work in making https://github.com/ampproject/amphtml/pull/32568 free of unrelated changes

**Pull request merged**